q2_fmt/_peds.py

```diff
@@ -466,11 +479,458 @@ def _create_masking(time_metadata, donor_df, recip_df, reference_column):
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        mismatched_peds = num_sum/donor_sum
+    mismatched_pairs_n = len(duplicated_recip_table.index)
+    per_sub_stats = _per_subject_stats(mismatched_peds,
+                                       actual_temp, iterations,
+                                       mismatched_pairs_n)
+    global_stats = _global_stats(per_sub_stats['p-value'])
+    return per_sub_stats, global_stats
+
+
+# peds_simulation helper functions
+
+def _create_mismatched_pairs(recip_df, metadata, used_references,
+                             reference_column):
+    """Creates a Dataframe of all the incorrect Donor-Recipient pairs
+
+    Creates a list of tuples of all donor-recipient pairs then filters out
+    any real donor-recipient pairs. The result is a Dataframe that includes
+    only mismatched donor-recipient pairs.
+
+    Parameters
+    ----------
+    recip_df: pd.DataFrame
+        A feature table of FMT recipients.
+    metadata: pd.DataFrame
+        The sample metadata for the study.
+    used_references: pd.Series
+        A series with recipients as the index and the associated reference
+        as the values.
+    reference_column: Str
+        Name of the reference column in the Sample Metadata.
+
+    Returns
+    -------
+    mismatched_df: pd.DataFrame
+        A DataFrame containing all mismatched pairs of donors and recipients.
+
+    Examples
+    --------
+    recip_df = pd.DataFrame({
+        'id': ['sample1', 'sample2', 'sample3'],
+        'Feature1': [1, 0, 0],
+        'Feature2': [0, 1, 0],
+        'Feature3': [0, 0, 1]}).set_index('id')
+
+    metadata_df = pd.DataFrame({
+        'id': ['sample1', 'sample2', 'sample3','donor1', 'donor2', 'donor3'],
+        'Ref': ['donor1', 'donor2', 'donor3', np.nan, np.nan, np.nan],
+        'subject': ['sub1', 'sub2', 'sub3', np.nan, np.nan, np.nan],
+        'group': [1, 1, 1, np.nan, np.nan, np.nan],
+        "Location": [np.nan, np.nan,np.nan, 'test', 'test','test']}
+        ).set_index('id')
+
+    used_references = pd.Series(data=['donor1', 'donor2', 'donor3'],
+                                    index=['sample1', 'sample2', 'sample3'],
+                                    name='Ref')
+    used_references.index.name = "id"
+
+    reference_column = "Ref"
+
+    _create_mismatched_pairs(recip_df, metadata_df, used_references,
+                             reference_column)
+    pd.DataFrame({'id': ["sample1", "sample1", "sample2", "sample2",
+                         "sample3", "sample3"],
+                  "Ref": ["donor2", "donor3", "donor1", "donor3",
+                          "donor1", "donor2"]}).set_index('id')
+    """
+    mismatched_pairs = []
+    for x in itertools.product(recip_df.index,
+                               metadata[reference_column].dropna().unique()):
+        mismatched_pairs.append(x)
+    mismatched_t = list(zip(used_references.index, used_references))
```

Comment:
I suggest calling this `matched_pairs`. 

You can also shuffle the order of the logic in this function to avoid having to create both the `mismatched_pairs` list and the `filtered` list, and to avoid one of your `for` loops. Create `matched_pairs` first,  and then iterate over the `product` (like you do now), but only add pairs to `mismatched_pairs` if they're not in `matched_pairs`.